### PR TITLE
User + Auth Optimizations

### PR DESCRIPTION
**Pull request by MacQSL**

### CODE REVIEW

**Review by NickPhura — APPROVED** [review]
🐟

Minus that user create sql issue we just discussed

File: src/utils/context.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+};
+
+/**
+ * Set the database context (used for row-level auditing).
+ *
+ * @async
+ * @param {DBContextConfig} config - Database context configuration
+ * @returns {Promise<void>}
+ */
+export const setDBContext = async (config: DBContextConfig) => {
```
[the remark this answers is not in the document]
Comment (NickPhura) [reply src/utils/context.ts:82]: Unless `txClient` already accounts for that? (Havent fully looked through all the changes yet)

[the remark this answers is not in the document]
Comment (MacQSL) [reply src/utils/context.ts:82]: Good idea, i'll update the comment.

File: src/services/user-service.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+     * are made at the same time. The setDatabaseUserContext will set the context for the incomming request,
+     * but if another request is made before the first one completes, the context will be overwritten.
+     * This will create issues with the audit columns.
+     */
+    try {
+      const user = await this.findUserByKeycloakUuid(payload.keycloak_uuid);
+
+      if (user) {
+        // User exists, set the context and return
+        // TODO: Drop this once the new design is implemented
```
Comment (NickPhura) [inline src/services/user-service.ts:154]: Still relevant?

Comment (NickPhura) [reply src/services/user-service.ts:154]: Does the context need to be set in this loginUser function at all, if it gets set in the transaction later on?

**Review by NickPhura — APPROVED** [review]
Looks pretty slick 🦆 

Tried a bunch of endpoints in sims, and they all seem happy. Code here looks good

**Pull request merged**